### PR TITLE
X16 board r2 - 4 joystick support

### PR DESCRIPTION
**Pull request by jjbliss**
This should have 4 joysticks working properly.
It has been tested with the [board r2 branch of the emulator](https://github.com/commanderx16/x16-emulator/tree/x16_board_r2) with my [test program](https://github.com/jjbliss/x16-controller-test)

### CODE REVIEW

File: kernal/drivers/x16/joystick.s [hunk 1]
```diff
@@ -101,26 +112,49 @@ l1:	lda nes_data
...
+	ldx joy3+1
+	ldy joy3+2
+	bra @5
 
-@2:	KVARS_END
+@4:
+	lda joy4
+	ldx joy4+1
+	ldy joy4+2
+	bra @5
```
Comment (greg-king5) [inline kernal/drivers/x16/joystick.s:154]: That branch isn't needed.

File: kernal/drivers/x16/joystick.s [hunk 2]
```diff
@@ -53,12 +58,18 @@ joystick_scan:
...
+	rol joy1,x ;If top bit was set, carry will be set rotate that into joy1
+	rol ;shift data to remove top bit.
+	cmp #top_bit ; Check top bit again (Controller #2)
+	rol joy2,x ;rol into joy2
+	rol
+	cmp #top_bit
+	rol joy3,x
+	rol
+	cmp #top_bit
+	rol joy4,x
```
Comment (greg-king5) [inline kernal/drivers/x16/joystick.s:71]: Those compares aren't needed.  Shift each accumulator bit into the carry flag, then rotate the flag into each joyN variable.

File: kernal/drivers/x16/joystick.s [hunk 3]
```diff
@@ -101,26 +112,49 @@ l1:	lda nes_data
 joystick_get:
 	KVARS_START
 	tax
-	bne @1       ; -> joy2
+	beq @1       ; -> joy1
+	cmp #1
+	beq @2       ; -> joy2
+	cmp #2
```
[the remark this answers is not in the document]
Comment (jjbliss) [reply kernal/drivers/x16/joystick.s]: I've updated this.

File: kernal/drivers/x16/joystick.s (hunk 1 above)
Comment (jjbliss) [reply kernal/drivers/x16/joystick.s]: I've removed it.

File: kernal/drivers/x16/joystick.s (hunk 2 above)
Comment (jjbliss) [reply kernal/drivers/x16/joystick.s:67]: I've updated this.

**Comment by jjbliss** [comment]
It's been quite a while since I put in this pull request.  I'm just starting to look at x16 stuff again.  the r2 board code hasn't been updated at all on GitHub since then.  Just wanted to check if people would still want this joystick support code added to this r2 branch of the rom or if the rom code has continued development elsewhere.

**Comment by mist64** [comment]
Thank you very much for this.

The whole X16 project had not made much progress for a while, mostly because the work on hardware was delayed. As long as we didn't have a working Proto #2 board, I did not want to merge the Proto #2 changes or anything related to it into master.

I now have a working Proto #2 board, so it is time!

https://www.commanderx16.com/forum/index.php?/topic/1210-proto-2-support-in-github-rom-emulator-documentation/

**Comment by mist64** [comment]
I also merged this into master.

**Pull request merged**